### PR TITLE
Task DEV-3

### PR DESCRIPTION
**Pull request by HancharykNastassia**

### CODE REVIEW

File: Fibonachchi.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
+﻿using System;
```
Comment (IgorKheidorov) [inline Fibonachchi.cs]: Please add other files necessary for project build

File: Fibonachchi.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+                        fib[2] = fib[0] + fib[1];
+                        f = (fib[2] == number);
+                    }
+                    if (f) Console.WriteLine(number.ToString() + " - число из последовательности Фибоначчи");
+                    else Console.WriteLine(number.ToString() + " - число не из последовательности Фибоначчи");
+                }                
+            }
+            catch (System.FormatException ex)
+            {
+                Console.WriteLine("Неверный формат");
```
Comment (IgorKheidorov) [inline Fibonachchi.cs]: Ask for re-input

File: Fibonachchi.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        static void Main (string [] args)
+        {
+            try
+            {
+                Console.WriteLine("Введите число");
+                int number = int.Parse(Console.ReadLine());
+                bool f = false;
+
+                if (number == 0 || number == 1)
+                Console.WriteLine(number.ToString() + " - число из последовательности Фибоначчи");
```
Comment (IgorKheidorov) [inline Fibonachchi.cs]: Code style

File: Fibonachchi.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+{
+    class FibonachchiMain
+    {
+        static void Main (string [] args)
+        {
+            try
+            {
+                Console.WriteLine("Введите число");
+                int number = int.Parse(Console.ReadLine());
+                bool f = false;
```
Comment (IgorKheidorov) [inline Fibonachchi.cs]: Please make meaningful name to 'f'

File: Fibonachchi.cs [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+                    int[] fib = new int[] { 0, 1, 1 };
+
+                    while (fib[2] < number)
+                    {
+                        fib[0] = fib[1];
+                        fib[1] = fib[2];
+                        fib[2] = fib[0] + fib[1];
+                        f = (fib[2] == number);
+                    }
+                    if (f) Console.WriteLine(number.ToString() + " - число из последовательности Фибоначчи");
```
Comment (IgorKheidorov) [inline Fibonachchi.cs]: Code style

File: Fibonachchi.cs [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+                int number = int.Parse(Console.ReadLine());
+                bool f = false;
+
+                if (number == 0 || number == 1)
+                Console.WriteLine(number.ToString() + " - число из последовательности Фибоначчи");
+                else
+                {
+                    int[] fib = new int[] { 0, 1, 1 };
+
+                    while (fib[2] < number)
```
Comment (IgorKheidorov) [inline Fibonachchi.cs]: Non-optimal cycle break condition, please think

**Comment by HancharykNastassia** [comment]
I really made some changes.

File: Fibonachchi.cs [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    class FibonachchiMain
+    {
+        static void Main (string [] args)
+        {
+            try
+            {
+                Console.WriteLine("Enter number");
+                int number = int.Parse(Console.ReadLine());
+                bool isfibonacci = false;
+                if ((number == 0) || (number == 1))
```
Comment (IgorKheidorov) [inline Fibonachchi.cs]: Code style, use {}brackets

File: Fibonachchi.cs [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+{
+    class FibonachchiMain
+    {
+        static void Main (string [] args)
+        {
+            try
+            {
+                Console.WriteLine("Enter number");
+                int number = int.Parse(Console.ReadLine());
+                bool isfibonacci = false;
```
Comment (IgorKheidorov) [inline Fibonachchi.cs]: better name according to code style  is isFibonacci

File: Fibonachchi.cs [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+                        isfibonacci = (fibonaccinumber[2] == number);
+                    }
+                    if (isfibonacci)
+                        Console.WriteLine(number.ToString() + " is in the Fibonacci sequence");
+                    else
+                        Console.WriteLine(number.ToString() + " is not in the Fibonacci sequence");
+                }                
+            }
+            catch (System.FormatException ex)
+            {
```
Comment (IgorKheidorov) [inline Fibonachchi.cs]: ask for re-enter number

File: Fibonachchi.sln [hunk 10]
```diff
@@ -0,0 +1,22 @@
+﻿
```
Comment (IgorKheidorov) [inline Fibonachchi.sln]: Solution one for all projects (tasks). One task one project, not solution

File: Fibonachchi.cs [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+            }
+            else
+            {
+                int[] fibonacciNumber = new int[] { 0, 1, 1 };
+                while (fibonacciNumber[2] < number)
+                {
+                    fibonacciNumber[0] = fibonacciNumber[1];
+                    fibonacciNumber[1] = fibonacciNumber[2];
+                    fibonacciNumber[2] = fibonacciNumber[0] + fibonacciNumber[1];
+                    isFibonacci = (fibonacciNumber[2] == number);
```
Comment (IgorKheidorov) [inline Fibonachchi.cs]: Not optima, break if ==, not wait for cycle end

**Review by IgorKheidorov — APPROVED** [review]

**Pull request merged**